### ISSUE TITLE
Support of data layers, polygons, GeoJSON in google maps plugin

### ISSUE THREAD

**Issue by sanekyy** [flutter/flutter#33514]
Hello. 

I can't find data layers features and opportunity to display GeoJson.

Here is descriptions of these features:

https://developers.google.com/maps/documentation/javascript/datalayer
https://github.com/googlemaps/android-maps-utils

Here is illustrations on these features:

![image](https://user-images.githubusercontent.com/18647037/58554245-2651c180-821f-11e9-99ee-1940f9512800.png)

![image](https://user-images.githubusercontent.com/18647037/58554268-3669a100-821f-11e9-9ebd-8fbfc66e3529.png)

Do you plan to add these features to the Google Map plugin?

**Comment by giaur500** [comment]
+1 I'm really missing that. Without geojson or even polygon support there is no any serious usage of this plugin possible

**Comment by otopba** [comment]
+1

**Comment by sanekyy** [comment]
Update:
I saw merged [pull request](https://github.com/flutter/plugins/commit/4900e75d147d52c150389c945fd516b69e5cf6b7)  with polygons support. Very cool!

Still needed GeoJson support.

**Comment by giaur500** [comment]
Well, geojson is actually json with polygons, markers and polylines definitions. Somebody needs to develop some parser.

**Comment by giaur500** [comment]
For geojson, I found this: https://github.com/Gubaer/dart-simple-features/blob/master/lib/src/geojson.dart

I have not tested yet, but it may be useful

**Comment by meshwara** [comment]
Is there any news for this GeoJSON ? Without geojson support, I can't porting my app with Flutter yet

**Comment by oceantechapp** [comment]
I load kml to polygon manually.
It seems very slow. Is anyway to make more efficency?

**Comment by iapicca** [comment]
Hi @sanekyy 
are you still experiencing this issue with the latest version of Flutter?

**Comment by giaur500** [comment]
I think still there is no geojson draw support in Google Maps Flutter, so issue is still valid.

**Comment by sanekyy** [comment]
@iapicca 
Hi. How is it related with Flutter version when this issue about google maps plugin?

**Comment by pixnbit** [comment]
Any update on this?

**Comment by HippyHawk** [comment]
https://pub.dev/packages/geojson
Here is a parser.

**Comment by taeilyoon** [comment]
how can i use it?

**Comment by giaur500** [comment]
Abandoned and it actually does not parse anything: https://github.com/synw/geojson/issues/30

No response, last update almost year ago. I think we can forget it.

**Comment by HippyHawk** [comment]
sorry for being so late, what ihave done is make a data model specific to my geoJson (use quicktype.io for data modeling).
then populate your data model and extract coordinates from it using forEach method. 
There is a class Polygons in google map plugin, you need to make a set of polygons and fill them with your extracted coordinates and initialize your google map with the set of polygons. Hope it helps

**Comment by draganjovanovic1** [comment]
Any update on this?

**Comment by timndichu** [comment]
Any enhancements yet?

**Comment by WaheedHussainHaans** [comment]
Any update on this?

**Comment by shapljts** [comment]
Any update on this?

**Comment by balaji101010** [comment]
Its been a year!! and still we stuck with this!..
Any update on this ?????

**Comment by yakupbaser** [comment]
any update?

**Comment by Shahmirzali-Huseynov** [comment]
any updates?

**Comment by giaur500** [comment]
No, still no support

**Comment by GULERTOLGA** [comment]
any update?

**Comment by lost2dreamer** [comment]
any updates ?

**Comment by mustafa-turkmen-alg** [comment]
any updates?

**Comment by giaur500** [comment]
For those who need geojson layers, clustering and other advanced features, I can recommend  FlutterMap: https://pub.dev/packages/flutter_map - geojson layers are supported with plugin: https://github.com/jozes/flutter_map_geojson

**Comment by mustafa-turkmen-alg** [comment]
Thank you for your feedback @giaur500 but I'm looking GeoJson integration for google maps sdk, The Flutter_map use mapview as tile view.  This block some feature of my app

**Comment by Keval184** [comment]
any updates folks ?

**Comment by giaur500** [comment]
Nope. Use 3rd party lib to parse geojson features and convert them to polygons you can draw on map. That's pretty easy imho

**Comment by stuartmorgan** [comment]
Please read https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#comments before commenting here.
